showpass/Showpass-E2E-tests
# TA-49 refactoring and code optimization

**Description:**
Here’s what will be changed as part of this user story:
- Add API request response code verification to the following method and reduce the implicit wait time cy.clickHeartIconToSaveEvents(1);
- Add a call to intercept the following request in the beforeEach loop /api/user/tickets/favorites/* of this spec cypress/e2e/user-settings/saved-events.cy.js
- Reduce or completely remove explicit waits
- Optimize the following failing method: verifyUpcomingPurchasedEventCard()

**Jira ticket:**
https://showpass.atlassian.net/browse/TA-49

**How to test:**
Run the following test suite and ensure all tests pass: cypress/e2e/user-settings/upcoming.cy.js

**Test execution results:**

<img width="748" alt="TA-49-all-tests-passed" src="https://github.com/showpass/Showpass-E2E-tests/assets/138615449/28597858-478f-4e6e-ae32-5edf087d2b9e">
